library/ssl_msg.c

```diff
@@ -4026,34 +4047,9 @@ int mbedtls_ssl_read_record( mbedtls_ssl_context *ssl,
             if( ret != 0 )
                 return( ret );
 
-            if( ssl_record_is_in_progress( ssl ) == 0 )
-            {
-#if defined(MBEDTLS_SSL_PROTO_DTLS)
-                int have_buffered = 0;
-
-                /* We only check for buffered messages if the
-                 * current datagram is fully consumed. */
-                if( ssl->conf->transport == MBEDTLS_SSL_TRANSPORT_DATAGRAM &&
-                    ssl_next_record_is_in_datagram( ssl ) == 0 )
-                {
-                    if( ssl_load_buffered_message( ssl ) == 0 )
-                        have_buffered = 1;
-                }
-
-                if( have_buffered == 0 )
-#endif /* MBEDTLS_SSL_PROTO_DTLS */
-                {
-                    ret = ssl_get_next_record( ssl );
-                    if( ret == MBEDTLS_ERR_SSL_CONTINUE_PROCESSING )
-                        continue;
-
-                    if( ret != 0 )
-                    {
-                        MBEDTLS_SSL_DEBUG_RET( 1, ( "ssl_get_next_record" ), ret );
-                        return( ret );
-                    }
-                }
-            }
+            ret = ssl_get_record_content( ssl );
+            if( ret != 0 )
+                continue;
```

Comment:
Behavior depending on the return value of `ssl_get_next_record`:

* `0`: before: go on; after: go on.
* `MBEDTLS_ERR_SSL_CONTINUE_PROCESSING`: before: `continue`; after; `continue`.
* `MBEDTLS_ERR_SSL_NON_FATAL`: can't happen because only `mbedtls_ssl_handle_message_type` returns this error.
* anything else: before: return; after: `continue`. This is ok because the while loop's condition makes the loop stop, and the subsequent `if( ret != 0 ) return( ret )` triggers.

This change removes debug messages conveying failures of `ssl_get_next_record` or `mbedtls_ssl_handle_message_type`. Both functions print their own debug message on error, or call a function that does. However they do so through many different code paths and it would be an easy mistake to miss one. So I'd prefer to keep these debug messages.